### PR TITLE
Add tests for join operation. Fix indexing upon initially setting MAGIC_NUMBERs

### PR DESCRIPTION
**Pull request by exfinen**
- Add tests for join operaiton
- Fix indexing upon initially setting MAGIC_NUMBERs
- Clean up test lib code

### CODE REVIEW

File: mpcstats_lib.py [hunk 1]
```diff
@@ -106,7 +105,7 @@ def join(data1: Matrix, data2: Matrix, data1_column_index: int, data2_column_ind
     # Initialize the rest of the matrix with MAGIC_NUMBER
     for i in range(num_columns_2):
         for j in range(num_rows_1):
-            new_data[num_columns_2 + i][j] = MAGIC_NUMBER
+            new_data[num_columns_1 + i][j] = MAGIC_NUMBER
```
Comment (mhchia) [inline mpcstats_lib.py:108]: good catch!

File: tests/test_ops.py [hunk 2]
```diff
@@ -128,3 +129,138 @@ def test_join_success():
...
+        1,
+        [
+            [0, 1, 2, 3],
+            [152, 160, 170, 180],
+            [M, M, M, M],
+            [M, M, M, M],
+        ],
+    )
+
+    # 4. No elems in col 1 mat 2 matches elems in col 0 mat 1
```
Comment (mhchia) [inline tests/test_ops.py:180]:
```suggestion
    # 4. Every element in col 0 mat 1 has a match in col 0 mat 2
```

File: tests/test_ops.py [hunk 3]
```diff
@@ -128,3 +129,138 @@ def test_join_success():
...
+        0,
+        0,
+        [
+            [0, 1, 2, 3],
+            [152, 160, 170, 180],
+            [0, M, 2, M],
+        ],
+    )
+
+    # 6. mat 1 has 1 column only <- fails
```
Comment (mhchia) [inline tests/test_ops.py:224]: why it fails? I think it should work as well

**Review by mhchia — APPROVED** [review]
Looks good to me. I've commented the comments. Should be ready if the comments are fixed

File: mpcstats_lib.py (hunk 1 above)
Comment (exfinen) [reply mpcstats_lib.py:108]: Great to confirm that!

File: tests/test_ops.py (hunk 3 above)
Comment (exfinen) [reply tests/test_ops.py]: I forgot to fix this comment. It's working as well. Sorry for the confusion. :)

**Pull request merged**